frontend/amundsen_application/static/js/ducks/search/sagas.ts

```diff
@@ -222,15 +231,44 @@ export function* searchResourceWorker(
   const { pageIndex, resource, term, searchType } = action.payload;
   const state = yield select(getSearchState);
   try {
-    const searchResults = yield call(
-      API.searchResource,
+    const response = yield call(
+      API.search,
       pageIndex,
-      resource,
+      RESULTS_PER_PAGE,
+      [resource],
       term,
-      state.filters[resource],
+      state.filters,
       searchType
     );
-    yield put(searchResourceSuccess(searchResults));
+
+    let searchResourceResults;
+    switch (resource) {
+      case ResourceType.table:
+        searchResourceResults = {
+          tables: response.table || initialState.tables,
+        };
+        break;
+      case ResourceType.user:
+        searchResourceResults = {
+          users: response.user || initialState.users,
+        };
+        break;
+      case ResourceType.dashboard:
+        searchResourceResults = {
+          dashboards: response.dashboard || initialState.dashboards,
+        };
+        break;
+      case ResourceType.feature:
+        searchResourceResults = {
+          features: response.feature || initialState.features,
+        };
+        break;
+      default:
+        searchResourceResults = {};
+    }
```

Comment:
I would extract this logic into a `computeSearchResourceResults(resource, response, initialState)` function call